packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -92,10 +94,26 @@ function useInsertionPoint( {
 				// Insert after a specific client ID.
 				_destinationIndex = getBlockIndex( clientId );
 			} else if ( ! isAppender && selectedBlockClientId ) {
```

Comment:
Just adding `__unstableGetEditorMode() !== 'zoom-out'` to this conditional is enough to get the right things you want. You don't need to set the `_destinationRootClientId = sectionRootClientId;` later on, as it should be the same or not matter.

Reply:
So this PR would just become 
```suggestion
			} else if ( ! isAppender && selectedBlockClientId && __unstableGetEditorMode() !== 'zoom-out') {
```

Reply:
Unless I'm confused, I don't think we can do that.

This code block needs to run even if not in zoom out. It's only setting `destination*` variables that's different in zoom out.